### PR TITLE
Show abbrv offers

### PR DESCRIPTION
**Pull request by sohanirao**

### CODE REVIEW

File: src/runtime/offers-flow-test.js [hunk 1]
```diff
@@ -14,11 +14,18 @@
  * limitations under the License.
  */
 
-import {ActivityPort} from 'web-activities/activity-ports';
+import {
+  ActivityPort,
+  ActivityResult,
+  ActivityResultCode} from 'web-activities/activity-ports';
```
Comment (dparikh) [inline src/runtime/offers-flow-test.js]: `ActivityResultCode` unused.

File: src/runtime/offers-flow.js [hunk 2]
```diff
@@ -140,3 +140,70 @@ export class SubscribeOptionFlow {
...
+
+    // If the user is already subscribed, trigger login flow
+    this.activityIframeView_.onMessage(data => {
+      if (data['alreadySubscribed'] && !data['loggedIn']) {
+        this.deps_.callbacks().triggerLoginRequest();
+        return;
+      }
+      // TODO(sohanirao) : Handle the case when user is logged in
+    });
+    console.log('view', this.activityIframeView_);
```
Comment (dparikh) [inline src/runtime/offers-flow.js]: Remove console.log

Here and line 200

File: src/runtime/runtime.js [hunk 3]
```diff
@@ -424,6 +430,14 @@ export class ConfiguredRuntime {
     });
   }
 
+  /** override */
+  showAbbrvOffer() {
+     return this.documentParsed_.then(() => {
```
Comment (dparikh) [inline src/runtime/runtime.js]: 2 chars indentation

File: src/runtime/runtime.js [hunk 4]
```diff
@@ -424,6 +430,14 @@ export class ConfiguredRuntime {
...
+       const flow = new AbbrvOfferFlow(this);
```
Comment (dparikh) [inline src/runtime/runtime.js]: Need to import AbbrvOffersFlow.
Can you test this in the UI?

You might want to expose this in the runtime under createPublicRuntime(), so can test in the UI.

File: src/runtime/offers-flow-test.js [hunk 5]
```diff
@@ -14,11 +14,17 @@
  * limitations under the License.
  */
 
-import {ActivityPort} from 'web-activities/activity-ports';
+import {
+  ActivityPort,
+  ActivityResult} from 'web-activities/activity-ports';
```
Comment (dvoytenko) [inline src/runtime/offers-flow-test.js]: nit: but do this:
```
import {
  ActivityPort,
  ActivityResult,
} from '....';
```

File: src/runtime/offers-flow-test.js [hunk 6]
```diff
@@ -14,11 +14,17 @@
...
+import {
+  acceptPortResult,
+} from './../utils/activity-utils';
 import {ConfiguredRuntime} from './runtime';
 import {
   OffersFlow,
   SubscribeOptionFlow,
+  AbbrvOfferFlow,
```
Comment (dvoytenko) [inline src/runtime/offers-flow-test.js]: Sort.

File: src/runtime/offers-flow-test.js [hunk 7]
```diff
@@ -193,3 +199,108 @@ describes.realWin('SubscribeOptionFlow', {}, env => {
     });
   });
 });
+
+describes.realWin('Abbrv Offer flow', {}, env => {
```
Comment (dvoytenko) [inline src/runtime/offers-flow-test.js]: `'AbbrvOfferFlow'` - b/c this is the most likely thing someone would search the tests for `AbbrvOfferFlow` class.

File: src/runtime/offers-flow-test.js [hunk 8]
```diff
@@ -193,3 +199,108 @@ describes.realWin('SubscribeOptionFlow', {}, env => {
...
+  });
+
+  it('should have valid AbbrvOfferFlow constructed', () => {
+    activitiesMock.expects('openIframe').withExactArgs(
+        sinon.match(arg => arg.tagName == 'IFRAME'),
+        '$frontend$/swg/_/ui/v1/abbrvofferiframe?_=_',
+        {
+          _client: 'SwG $internalRuntimeVersion$',
+          publicationId: 'pub1',
+          productId: 'pub1:label1',
```
Comment (dvoytenko) [inline src/runtime/offers-flow-test.js]: Do we really need `productId`?

File: src/runtime/offers-flow-test.js [hunk 9]
```diff
@@ -193,3 +199,108 @@ describes.realWin('SubscribeOptionFlow', {}, env => {
...
+
+  it('should trigger login flow for a subscribed user', () => {
+    let messageCallback;
+    sandbox.stub(port, 'onMessage', callback => {
+      messageCallback = callback;
+    });
+    const loginStub = sandbox.stub(runtime.callbacks(), 'triggerLoginRequest');
+    activitiesMock.expects('openIframe').returns(Promise.resolve(port));
+    return abbrvOfferFlow.start().then(() => {
+      messageCallback({'alreadySubscribed': true, 'loggedIn': false});
```
Comment (dvoytenko) [inline src/runtime/offers-flow-test.js]: Let's rename to "linkRequested" right away.

File: src/runtime/offers-flow-test.js [hunk 10]
```diff
@@ -193,3 +199,108 @@ describes.realWin('SubscribeOptionFlow', {}, env => {
...
+
+  it('should not trigger login flow for subscibed logged in user', () => {
+    let messageCallback;
+    sandbox.stub(port, 'onMessage', callback => {
+      messageCallback = callback;
+    });
+    const loginStub = sandbox.stub(runtime.callbacks(), 'triggerLoginRequest');
+    activitiesMock.expects('openIframe').returns(Promise.resolve(port));
+    return abbrvOfferFlow.start().then(() => {
+      messageCallback({'alreadySubscribed': true, 'loggedIn': true});
```
Comment (dvoytenko) [inline src/runtime/offers-flow-test.js]: Ditto, rename "loggedIn"

File: src/runtime/offers-flow.js [hunk 11]
```diff
@@ -140,3 +140,70 @@ export class SubscribeOptionFlow {
...
+    /** @private @const {!../components/dialog-manager.DialogManager} */
+    this.dialogManager_ = deps.dialogManager();
+
+    /** @private @const {!ActivityIframeView} */
+    this.activityIframeView_ = new ActivityIframeView(
+        this.win_,
+        this.activityPorts_,
+        feUrl('/abbrvofferiframe'),
+        feArgs({
+          'productId': deps.pageConfig().getProductId(),
```
Comment (dvoytenko) [inline src/runtime/offers-flow.js]: Remove.

File: src/runtime/offers-flow.js [hunk 12]
```diff
@@ -140,3 +140,70 @@ export class SubscribeOptionFlow {
...
+
+  /**
+   * Starts the offers flow
+   * @return {!Promise}
+   */
+  start() {
+
+    // If the user is already subscribed, trigger login flow
+    this.activityIframeView_.onMessage(data => {
+      if (data['alreadySubscribed'] && !data['loggedIn']) {
```
Comment (dvoytenko) [inline src/runtime/offers-flow.js]: Let's rename to "linkRequested" and it should be more like this:
```
      if (result['alreadySubscribed']) {
        this.deps_.callbacks().triggerLoginRequest({
          linkRequested: !!result['linkRequested'],
        });
        return;
      }
```

Basically it's always login, but not always link.

File: src/runtime/offers-flow.js [hunk 13]
```diff
@@ -140,3 +140,70 @@ export class SubscribeOptionFlow {
...
+        this.deps_.callbacks().triggerLoginRequest({
+          linkRequested: true,
+        });
+        return;
+      }
+      // TODO(sohanirao) : Handle the case when user is logged in
```
Comment (dvoytenko) [inline src/runtime/offers-flow.js]: Remove

File: src/runtime/runtime.js [hunk 14]
```diff
@@ -29,6 +29,7 @@ import {OffersApi} from './offers-api';
 import {
   OffersFlow,
   SubscribeOptionFlow,
+  AbbrvOfferFlow,
```
Comment (dvoytenko) [inline src/runtime/runtime.js]: Sort

File: src/runtime/offers-flow-test.js (hunk 5 above)
Comment (sohanirao) [reply src/runtime/offers-flow-test.js]: Done

File: src/runtime/offers-flow-test.js (hunk 6 above)
Comment (sohanirao) [reply src/runtime/offers-flow-test.js]: Done

File: src/runtime/offers-flow-test.js (hunk 7 above)
Comment (sohanirao) [reply src/runtime/offers-flow-test.js]: Done

**Pull request merged**